File: Ahorn/entities/AbyssalEye.jl
module AbyssalEye

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/AbyssalEye" AbyssalEye(x::Integer, y::Integer, gazeRange::Integer=200, gazeWidth::Integer=30, health::Integer=2)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(AbyssalEye)
)

function Ahorn.selection(entity::AbyssalEye)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::AbyssalEye, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end

File: framework/extracted/Loenn/entities/abyssalEye.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local abyssalEye = {}

abyssalEye.name = "MaggyHelper/AbyssalEye"
abyssalEye.depth = -100
abyssalEye.fieldInformation = {
    gazeRange = {
        minimumValue = 0
    },
    gazeWidth = {
        minimumValue = 0
    },
    health = {
        minimumValue = 1
    }
}
abyssalEye.placements = {
    {
        name = "normal",
        data = {
            gazeRange = 200,
            gazeWidth = 30,
            health = 2
        }
    }
}

function abyssalEye.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("characters/abyssal_eye/dormant", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 0.5)
    sprite.color = {0.5, 0.0, 0.0, 1.0}
    return sprite
end

function abyssalEye.selection(room, entity)
    return utils.rectangle(entity.x - 16, entity.y - 16, 32, 32)
end

return abyssalEye
